### ISSUE TITLE
illuminate.nvim does not refresh highlights after autojump

### ISSUE THREAD

**Issue by xzbdmw** [ggandor/leap.nvim#221]
I have illuminate.nvim and it will highlight cursorword, but leap's autojump seems not trigger their detect of new cursorpos, only after I type the third char. Is it probably a fake cursor during autojump?

https://github.com/ggandor/leap.nvim/assets/97848247/458d575e-7564-439a-8119-3137bfe58670

**Comment by xzbdmw** [comment]
nightly

**Comment by ggandor** [comment]
Does this snippet fix the issue?
```lua
vim.api.nvim_create_augroup('LeapIlluminate', {})
vim.api.nvim_create_autocmd('User', {
  pattern = 'LeapPatternPost',
  callback = function ()
    vim.api.nvim_exec_autocmds('CursorMoved', {
      group = 'vim_illuminate_v2_augroup'
    })
  end,
  group = 'LeapIlluminate',
})
```

**Comment by ggandor** [comment]
> Is it probably a fake cursor during autojump?

No, on nightly it's the real one.

**Comment by ggandor** [comment]
You might ask this in the illuminate repo too.

**Comment by xzbdmw** [comment]
Hi, if I use this function
```lua
vim.api.nvim_create_augroup("LeapIlluminate", {})
vim.api.nvim_create_autocmd("User", {
    pattern = "LeapPatternPost",
    callback = function()
        local winid = vim.api.nvim_get_current_win()
        local cursor = vim.api.nvim_win_get_cursor(winid)
        -- __AUTO_GENERATED_PRINT_VAR_START__
        print([==[function cursor:]==], vim.inspect(cursor)) -- __AUTO_GENERATED_PRINT_VAR_END__
        require("illuminate.engine").refresh_references(_, _, "hi")
    end,
    group = "LeapIlluminate",
})
```
the cursor pos is not updated when autojump, so illuminate can't update too

**Comment by ggandor** [comment]
Sorry, try `LeapSelectPre` as the pattern (`PatterPost` is before the jump).

**Comment by xzbdmw** [comment]
If I add a redraw command under https://github.com/RRethy/vim-illuminate/blob/305bf07b919ac526deb5193280379e2f8b599926/lua/illuminate/highlight.lua#L51
it works

**Comment by jamestansx** [comment]
Had the same problem too, but I didn't have `vim-illuminate` installed. Bisecting shows that `74b2ed7` introduces this bug

**Comment by jamestansx** [comment]
The reason is that the fix required has been reverted on https://github.com/neovim/neovim/pull/28235

and now that `nvim__redraw` has been merged on nightly though it's experimental (https://github.com/neovim/neovim/pull/28101), calling `nvim__redraw` on `LeapSelectPre` should work.

**Comment by einCyberSimon** [comment]
Any updates on this one? I also have this problem.
I cannot use the default mappings as they conflict with [gbprod/substitute.nvim](https://github.com/gbprod/substitute.nvim)
Remapping the keys with
```lua
keymap.set({ "n", "x", "o" }, "f", "<Plug>(leap-forward)")
keymap.set({ "n", "x", "o" }, "F", "<Plug>(leap-backward)")
keymap.set({ "n", "x", "o" }, "gf", "<Plug>(leap-from-window)")
```
results in the aforementioned issue.
This happens both on v10 and nightly.

**Comment by ggandor** [comment]
@einCyberSimon I cannot reproduce, both 0.10 stable and nightly 0.11 should work. (Fixed by f1f19fc, it has nothing to do with the mappings.)

**Comment by einCyberSimon** [comment]
Hm, there seems to be another problem, as jumping does not work at all anymore (also with `require("leap").create_default_mappings()`, even with resetting my config and reinstalling nvim.
My config currently looks like this (using lazy.nvim)
```lua
return {
    "ggandor/leap.nvim",
    dependencies = { "tpope/vim-repeat" },
    config = function()
        require("leap").create_default_mappings()
    end,
}
```
After pressing `s`, than some character (for instance `c`) and a second character (e.g., `l`), it says `not found: cl`

**Comment by ggandor** [comment]
@einCyberSimon If the problem persists, please open a separate issue.
